internal/cmd/local/local/install.go

```diff
@@ -378,6 +380,63 @@ func (c *Command) watchEvents(ctx context.Context) {
 	}
 }
 
+// 2024-09-10 20:16:24 WARN i.m.s.r.u.Loggers$Slf4JLogger(warn):299 - [273....
+var javaLogRx = regexp.MustCompile(`^\d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2} \x1b\[(?:1;)?\d+m(?P<level>[A-Z]+)\x1b\[m (?P<msg>\S+ - .*)`)
+
+func (c *Command) streamPodLogs(ctx context.Context, namespace, podName, prefix string, since time.Time) error {
+	r, err := c.k8s.StreamPodLogs(ctx, airbyteNamespace, airbyteBootloaderPodName, since)
+	if err != nil {
+		return err
+	}
+	defer r.Close()
+
+	level := pterm.Debug
+	scanner := bufio.NewScanner(r)
+
+	for scanner.Scan() {
```

Comment:
Can we check the state of the `ctx` in this loop? Break out if the context is no longer active?

Reply:
I proved this to myself. I couldn't think of a nice way to write an actual test for it (since it depends on the real k8s implementation), but I hacked in this code (note `javaLogScanner` is not in this PR, but it's just a scanner with the java parsing wrapped in)

```
func (c *Command) streamPodLogs(ctx context.Context, namespace, podName, prefix string, since time.Time) error {

	ctx, cancel := context.WithCancel(ctx)
	go func() {
		log.Printf("sleeping\n")
		time.Sleep(30 * time.Second)
		log.Printf("canceling\n")
		cancel()
	}()

	r, err := c.k8s.StreamPodLogs(ctx, namespace, podName, since)
	if err != nil {
		return err
	}
	defer r.Close()

	s := newJavaLogScanner(r)
	for s.Scan() {
		if s.line.level == "ERROR" {
			pterm.Error.Printfln("%s: %s", prefix, s.line.msg)
		} else {
			pterm.Debug.Printfln("%s: %s", prefix, s.line.msg)
		}
	}

	log.Printf("scanner exited: %v", s.Err())
	return s.Err()
}
```

and you can see the scanner exists in the logs:
<img width="992" alt="image" src="https://github.com/user-attachments/assets/94cd48a0-4be5-4509-88be-4b07ef3afee7">